BurstBotNET/Commands/ChaseThePig/Join.cs

```diff
@@ -0,0 +1,224 @@
+using BurstBotShared.Shared.Extensions;
+using BurstBotShared.Shared.Interfaces;
+using BurstBotShared.Shared.Models.Game.ChaseThePig;
+using BurstBotShared.Shared.Models.Game.ChaseThePig.Serializables;
+using BurstBotShared.Shared.Models.Game.Serializables;
+using BurstBotShared.Shared.Utilities;
+using Microsoft.Extensions.Logging;
+using Remora.Discord.API.Abstractions.Objects;
+using Remora.Results;
+
+namespace BurstBotNET.Commands.ChaseThePig;
+
+using ChasePigGame = IGame<ChasePigGameState, RawChasePigGameState, ChaseThePig, ChasePigPlayerState, ChasePigGameProgress, ChasePigInGameRequestType>;
+
+public partial class ChaseThePig
+{
+    private async Task<IResult> Join(float baseBet, params IUser?[] users)
+    {
+        var joinResult = await Game.GenericJoinGame(
+            baseBet, users, GameType.ChaseThePig, "/chase_the_pig/join",
+            _state, _context, _interactionApi, _userApi, _logger);
+
+        if (joinResult == null) return Result.FromSuccess();
+
+        switch (joinResult.JoinStatus.StatusType)
+        {
+            case GenericJoinStatusType.Start:
+            {
+                var result = await Game.GenericStartGame(
+                    _context, joinResult.Reply, joinResult.InvokingMember, joinResult.BotUser,
+                    joinResult.JoinStatus, GameName, "/chase_the_pig/join/confirm",
+                    joinResult.MentionedPlayers.Select(s => s.Value),
+                    _state, 4, _interactionApi, _channelApi,
+                    _guildApi, _logger);
```

Comment:
That is a lot of arguments 😅 Maybe you can simplify this in some way? Or used named arguments

Reply:
Rider would suggest against redundant named arguments though 🤔 https://www.jetbrains.com/help/rider/ArgumentsStyleNamedExpression.html

Reply:
I made a new record type to refactor it out